js/source/worker_tile.js

```diff
@@ -108,9 +108,9 @@ WorkerTile.prototype.parse = function(data, layerFamilies, actor, rawTileData, c
         }
     }
 
-    var buckets = [],
+    var buckets = this.buckets = [],
```

Comment:
Agreed.  I think [this line](https://github.com/mapbox/mapbox-gl-js/blob/96c010c3c4b2e87afcc669275e32b1a458f42f56/js/source/worker_tile.js#L195) must certainly help with that, but we should also investigate whether the arraybuffers continue to consume memory after they're transferred/neutered.  If not, or if there's a straightforward way to drop them after we've made the transfer, then probably the memory impact here is minimal?

However, if not:
 - I _think_ we could hone this down to only storing the feature-index->array-range mapping for each bucket, although that would require us to recreate the Bucket objects in `updateProperties`, and I would imagine there would be a few wrinkles there.
 - Alternatively, we could _not_ hold onto `this.buckets` by default, but only do so if a flag is set in the params